src/main/kotlin/ch/hevs/cloudio/cloud/restapi/endpoint/permission/EndpointPermissionController.kt

```diff
@@ -63,13 +59,9 @@ class EndpointPermissionController(
         }.id.let { groupID ->
             when {
                 permission == EndpointPermission.DENY -> userGroupEndpointPermissionRepository.deleteByUserGroupIDAndEndpointUUID(groupID, uuid)
-                permission.fulfills(EndpointPermission.WRITE) -> {
-                    val endpointPermission = userGroupEndpointPermissionRepository.findByUserGroupIDAndEndpointUUID(groupID, uuid).orElse(UserGroupEndpointPermission(groupID, uuid, permission))
-                    endpointPermission.modelPermissions.clear()
-                    userGroupEndpointPermissionRepository.save(endpointPermission)
-                }
                 else -> {
                     val endpointPermission = userGroupEndpointPermissionRepository.findByUserGroupIDAndEndpointUUID(groupID, uuid).orElse(UserGroupEndpointPermission(groupID, uuid, permission))
+                    endpointPermission.permission = permission
```

Comment:
We should still call `endpointPermission.modelPermissions.clear()` to get rid of model permission as they are not longer needed.